pkg/apis/testing/v1alpha1/testsuite_types.go

```diff
@@ -0,0 +1,138 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TestSuiteSpec defines the desired state of TestSuite
+type TestSuiteSpec struct {
+	// How many tests we want to execute at the same time.
+	// Depends on cluster size and it's load
+	ConcurrencyLevel int64 `json:"concurrencyLevel,omitempty"`
+	// You can specify test names explicitly.
+	TestNamesSelector []NamespacedTest `json:"testNamesSelector,omitempty"`
+	// Will run every test that depends of any of the component listed here
+	ComponentsSelector []string `json:"componentsSelector,omitempty"`
```

Comment:
I assume TestNamesSelector and ComponentsSelector are exclusive?

Reply:
IMO you can define both of them at the same time. You can define to run all tests for "service-catalog" and this specific test for kubeless component. Thanks to that our API is more flexible.